### PR TITLE
add analog waveform support and partially fix decimation for charts

### PR DESCRIPTION
**Pull request by cipix2000**

### CODE REVIEW

File: jquery.flot.charting.js [hunk 1]
```diff
@@ -8,42 +8,82 @@ Licensed under the MIT license.
...
+        var index = plot.getData().indexOf(series);
+        if (index < historyBuffer.width) {
+            var data = series.historyBuffer.query(start, end, step, index);
+        } else {
+            data = [];
+        }
+
+        var points = datapoints.points;
+        for (var i = 0; i < data.length; i+=2) {
+            points[i] = indexMap ? indexMap[data[i]] : data[i];
```
Comment (cristiingineru) [inline jquery.flot.charting.js]: How long is a typical data? Does it make sense to have 2 versions of this for statement? One for indexMap !== and another one for indexMap === undefined

File: jquery.flot.charting.js [hunk 2]
```diff
@@ -8,42 +8,82 @@ Licensed under the MIT license.
...
+
+        points.length = data.length;
+        datapoints.pointsize = 2;
+
+        return points;
+    }
+
     // remove old data series and trigger the computation of a new one from the history buffer
     function updateSeries(plot, historyBuffer) {
         var dataSeries = plot.getData();
```
Comment (cristiingineru) [inline jquery.flot.charting.js]: In the decimateChartData() function it looked like it is possible to have more series than the width of the historyBuffer. Here it looks like only the first historyBuffer.width are updated.

File: jquery.flot.charting.js [hunk 3]
```diff
@@ -8,42 +8,82 @@ Licensed under the MIT license.
 (function ($) {
     'use strict';
 
+    var plot;
```
Comment (cristiingineru) [inline jquery.flot.charting.js]: Do we have any test with multiple plots being tested in the same time?

Comment (cipix2000) [inline jquery.flot.charting.js]: fuck! good catch. No we don't have tests with multiple charts

File: jquery.flot.segment-tree.js [hunk 4]
```diff
@@ -326,13 +328,15 @@ Licensed under the MIT license.
...
+                data.push(i);
+                data.push(this.get(i));
+
             }
         } else {
             var minmax = new SegmentTreeNode();
 
             var maxIndex, minIndex;
-            for (i = start; i < end; i += step) {
+            for (i = start; i < end; i = Math.round( i + step)) {
```
Comment (cristiingineru) [inline jquery.flot.segment-tree.js]: Is it possible for this loop to run infinitely when ```Math.round(i+step) == i``` or ```step < 0.5``` ?

File: spec/historyBuffer_aw.Test.js [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+
+        expect(hb.toArray()).toEqual([[aw, aw1], [aw2, aw3]]);
+    });
+
+    it('has a toDataSeries method', function () {
+        var hb = new HistoryBuffer_aw(10);
+
+        hb.appendArray([aw, aw1]);
+
+        expect(hb.toDataSeries(0)).toEqual([[4, 1], [5, 2], [6, 3], [null, null], [1, 1], [2, 2], [3, 3]]);
```
Comment (cristiingineru) [inline spec/historyBuffer_aw.Test.js]: Could you please use different values for aw and aw1 to make the pattern more obvious.

File: jquery.flot.segment-tree.js (hunk 4 above)
Comment (cipix2000) [inline jquery.flot.segment-tree.js]: no. above in the code we make sure that for steps smaller than 4 we use a different code path

File: jquery.flot.charting.js (hunk 1 above)
Comment (cipix2000) [inline jquery.flot.charting.js]: I hope to get rid of the index map soon

**Pull request closed without merging**